### PR TITLE
Feat: deploy scripts and deploy contracts to Mode and Optimism

### PR DESCRIPTION
**Pull request by codebuster22**

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -0,0 +1,8 @@
+L2_BRIDGE_ADDRESS=0x4200000000000000000000000000000000000010
```
Comment (njokuScript) [inline .env.example:1]: is this same across the superchain?

File: README.md [hunk 2]
```diff
@@ -1,66 +1,42 @@
-## Foundry
+## SuperMigrate
```
Comment (njokuScript) [inline README.md]: Supermigrate

File: src/migration/factory/BasedERC20Factory.sol [hunk 3]
```diff
@@ -14,6 +14,7 @@ import {Clones} from "@openzeppelin/contracts/proxy/Clones.sol";
 /**
  * @title BasedERC20Factory
  * @notice Factory contract for creating and deploying BasedMigrateERC20 tokens.
```
Comment (njokuScript) [inline src/migration/factory/BasedERC20Factory.sol]: can we change `BasedMigrateERC20` to `SupermigrateERC20`? This way we're not just agnostic to Base but to all L2s

**Review by njokuScript — CHANGES_REQUESTED** [review]
Look at the comments I left @codebuster22 

Great work, btw!

File: .env.example (hunk 1 above)
Comment (codebuster22) [reply .env.example:1]: Yes, Though, scroll is not part of Optimism Super Chain.

File: README.md (hunk 2 above)
Comment (codebuster22) [reply README.md]: cool.

File: src/migration/factory/BasedERC20Factory.sol (hunk 3 above)
Comment (codebuster22) [reply src/migration/factory/BasedERC20Factory.sol]: Yes, makes sense.

**Review by njokuScript — APPROVED** [review]

**Pull request merged**